src/compiler/utilities.ts

```diff
@@ -4581,6 +4587,23 @@ namespace ts {
         return (declaration as NamedDeclaration).name;
     }
 
+    function getAssignedName(node: Node): DeclarationName {
+        if (!node.parent) {
+            return undefined;
+        }
+        else if (isPropertyAssignment(node.parent)) {
+            return node.parent.name;
+        }
+        else if (isBinaryExpression(node.parent) && node === node.parent.right) {

```

Comment:
No love for binding pattern initializers?:
```ts
const {
a = function() {},
b = class {}
} = ({} as any);
```

Reply:
Done! (But, no, I don't like binding pattern initializers)